### DISCUSSION TITLE
Facing RangeError When Applying Filters to Large Images in Fabric.js Canvas with textureSize Set

### DISCUSSION THREAD

**Discussion by Arkit-Sutariya** [fabricjs/fabric.js#10045] [Q&A]
I'm using Fabric.js version 5.3.0, applying filters to image objects with the function below. However, I'm encountering an error: "ERROR RangeError: Invalid typed array length: 20736000." This error occurs only when the image size exceeds 2048px

I have initialized my Fabric and canvas as follows.
----------------------------------------------------------------------------------------------
import * as latestFabric from 'fabric-latest';

 ngOnInit() {
       this.canvas = new latestFabric.fabric.Canvas('canvas');
 }
 
setFilter(filtertype, value, filterOpacity) {
        if (this.canvas.getActiveObject() && this.canvas.getActiveObject().type === 'image') {
            latestFabric.fabric.filterBackend = new latestFabric.fabric.WebglFilterBackend();
            latestFabric.fabric.textureSize = 8192;
            latestFabric.fabric.isWebglSupported(latestFabric.fabric.textureSize);
        }
        switch (filtertype) {
            case 'brightness':
                var filter = new latestFabric.fabric.Image.filters.Brightness({
                    brightness: (this.checkLimitBeforeApplyForInputSlider(Math.round(value), 100, 0) / 250)
                });
                this.rangeMoveTimeout = setTimeout(() => {
                    this.canvas.getActiveObject().applyFilters();
                    this.canvas.renderAll();
                }, 500);
                break;

            case 'contrast':
                var filter = new latestFabric.fabric.Image.filters.Contrast({
                    contrast: (this.checkLimitBeforeApplyForInputSlider(Math.round(value), 100, 0) / 250)
                });
                this.rangeMoveTimeout = setTimeout(() => {
                    this.canvas.getActiveObject().applyFilters();
                    this.canvas.renderAll();
                }, 500);
                break;
                }
   }

Due to certain restrictions, I am unable to provide you with code in any editor, such as jsfiddle.net.
I've tested the same scenario on different PCs, and it works without error on higher-spec PCs, but the cut image appears.

**Comment by asturur** [comment]
What is the maxTextureSize of the machine that fails?

**Reply by Arkit-Sutariya** [reply]
Hey @asturur 

Thanks for your response...

On the PC where I'm getting the error, the max texture size is 8192. On another PC, where there's no error but the image appears cut off, the max texture size is 16384.

**Reply by asturur** [reply]
you should pinpoint the exact line where that error happens and the image size at that moment.
Could be that you fail at allocate a 20 meg array or that is not good for that width/height of the image

**Reply by Arkit-Sutariya** [reply]
the error occurs at this line: **this.canvas.getActiveObject().applyFilters();** when I apply a filter to the active object. I suspect it's due to memory limitations rather than unsuitable image dimensions, as it works on higher configuration PCs.

**Reply by asturur** [reply]
Could it be you are out of memory in that machine? How many images are you loading? You should profile the operation with the dev tools and see memory allocation
